packages/block-library/src/audio/edit.js

```diff
@@ -31,23 +30,22 @@ import { createUpgradedEmbedBlock } from '../embed/util';
 
 const ALLOWED_MEDIA_TYPES = [ 'audio' ];
 
-class AudioEdit extends Component {
-	constructor() {
-		super( ...arguments );
-		this.toggleAttribute = this.toggleAttribute.bind( this );
-		this.onSelectURL = this.onSelectURL.bind( this );
-		this.onUploadError = this.onUploadError.bind( this );
-	}
+function AudioEdit( {
+	attributes,
+	noticeOperations,
+	setAttributes,
+	onReplace,
+	isSelected,
+	noticeUI,
+} ) {
+	const { id, autoplay, caption, loop, preload, src } = attributes;
 
-	componentDidMount() {
-		const {
-			attributes,
-			mediaUpload,
-			noticeOperations,
-			setAttributes,
-		} = this.props;
-		const { id, src = '' } = attributes;
+	const mediaUpload = useSelect( ( select ) => {
+		const { getSettings } = select( 'core/block-editor' );
+		return getSettings().mediaUpload;
+	} );
```

Comment:
You still need to add a dependency array here to memoize the `useSelect` callback.
```suggestion
	}, [] );
```

Reply:
Oh yes I don't understand on this way. I fix it now.